Feature: bob sync (and `sync_version_file_path` config)

Avishayy opened the pull request:
#80 

Using `bob sync` will try to read a version stored at the configured `sync_version_file_path`. This is useful when using version control to track local configurations, where one might want to document when the version of neovim was changed (and to what version).

Note that this **does not** replace the `used` file in the downloads directory, `used` is regarded as the truth on the current installed neovim version, whereas `sync_version_file_path` can have any content, regardless of the current installed version.

MordechaiHadad commented:
I think the name of `used_version_file_path` should change to something else, can be confused with the original used file.
Maybe `sync_version_file_path` or something idk for sure.

Avishayy commented:
Sounds good to me, I renamed all instances to `sync_version_file_path`

MordechaiHadad commented:
also another point add this sync path thingy to the env var handler here: https://github.com/MordechaiHadad/bob/pull/78/files#diff-42cb6807ad74b3e201c5a7ca98b911c5fa08380e942be6e4ac5807f8377f87fcR54

Avishayy commented:
Done, note that the current env var implementation will crash when the config doesn't have an env var [as I commented](https://github.com/MordechaiHadad/bob/pull/78/files#r1070577678)

MordechaiHadad commented:
btw check this review thingy i opened so i can merge this sooner.

MordechaiHadad commented:
oh shit it closed this PR when i merged the previous PR

Avishayy commented:
I rebased over the latest main, let me know if you have any more comments

MordechaiHadad commented:
> I rebased over the latest main, let me know if you have any more comments

Only need you to address the review and you are practically done, i think.

Avishayy commented:
> > I rebased over the latest main, let me know if you have any more comments
> 
> Only need you to address the review and you are practically done, i think.

I addressed everything you mentioned, renaming it to `sync_version_file_path` and supporting env vars in the config, have I missed anything else?

MordechaiHadad commented:
weird... can't you  see this review i have opened?
![wwXshJmNHG](https://user-images.githubusercontent.com/33547558/212708089-fa0cc365-a437-410a-bfbd-3756dea6d9ba.png)

Avishayy commented:
> weird... can't you see this review i have opened? ![wwXshJmNHG](https://user-images.githubusercontent.com/33547558/212708089-fa0cc365-a437-410a-bfbd-3756dea6d9ba.png)

No.

See that your comment is marked as `pending`, you need to submit the review in the "Files Changed" tab at the top right
![image](https://user-images.githubusercontent.com/5567310/212708417-ce746a02-b234-42b2-95a6-341e5cc5ece0.png)
